SwanProjects/swanprojects/utils.py

```diff
@@ -0,0 +1,233 @@
+            f"source '{setup_script_path}' && printenv",
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+            env={
+                "RELEASE": stack.get("release"),
+                "PLATFORM": stack.get("platform"),
+                "SWAN_PROJECT_PATH": os.path.join(
+                    os.path.abspath(self.root_dir), project_folder_path
+                ),
+                **pass_through_env,
+            },
+        )
+        stdout, stderr = await process.communicate()
+        if process.returncode != 0:
+            self.log.error(
+                f"RETURNCODE: {process.returncode}, STDOUT:\n {stdout}\n, STDERR: \nf{stderr}\n"
+            )
+            raise Exception(
+                f"Error generating environment with output {stdout}, {stderr}, {process.returncode}"
+            )
+        self._save_environment_lock_file(project_folder_path, stdout.decode("utf-8"))
+
+    def _save_environment_lock_file(self, project_folder_path: str, environment: str):
+        environment_file_path = os.path.join(
+            project_folder_path, self.swan_config.environment_file_name
+        )
+        model = {
+            "name": self.swan_config.environment_file_name,
+            "path": environment_file_path,
+            "type": "file",
+            "content": environment,
+            "format": "text",
+        }
+        self.contents_manager.save(model, environment_file_path)
+
+    def _save_enviornment_metadata_file(self, project_folder_path: str, stack: Any):
+        metadata_file_path = os.path.join(
+            project_folder_path, self.swan_config.project_file_name
+        )
+        if stack["type"] == "default":
+            if  self.contents_manager.file_exists(metadata_file_path):
+                self.contents_manager.delete_file(metadata_file_path)
+        else:
+            model = {
+                "name": self.swan_config.project_file_name,
+                "path": metadata_file_path,
+                "type": "file",
+                "content": json.dumps({"stack": stack}),
+                "format": "text",
+            }
+            self.contents_manager.save(model, metadata_file_path)
+
+    def _save_user_script_file(self, project_folder_path: str, user_script: str):
+        user_script_file_path = os.path.join(
+            project_folder_path, self.swan_config.project_file_name
+        )
+        if not user_script and self.contents_manager.file_exists(user_script_file_path):
+            # If the script was removed, delete any existing file
+            self.contents_manager.delete_file(user_script_file_path)
+        else:
+            model = {
+                "name": self.swan_config.project_file_name,
+                "path": user_script_file_path,
+                "type": "file",
+                "content": user_script,
+                "format": "text",
+            }
+            self.contents_manager.save(model, user_script_file_path)
```

Comment:
Why do you use this method? There're other, lower level ones, that you can use and don't require you to set all of this model stuff.
(same applies to other places you use it)